x-pack/plugins/search_indices/public/components/indices/details_page.tsx

```diff
@@ -5,25 +5,151 @@
-import React, { useCallback, useMemo } from 'react';
+import {
+  EuiPageSection,
+  EuiSpacer,
+  EuiButton,
+  EuiPageTemplate,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiPopover,
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiText,
+  EuiIcon,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import React, { useCallback, useMemo, useState } from 'react';
 import { useParams } from 'react-router-dom';
 import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { SectionLoading } from '@kbn/es-ui-shared-plugin/public';
 import { useIndex } from '../../hooks/api/use_index';
 import { useKibana } from '../../hooks/use_kibana';
+import { DeleteIndexModal } from './delete_index_modal';
 
 export const SearchIndexDetailsPage = () => {
   const indexName = decodeURIComponent(useParams<{ indexName: string }>().indexName);
-  const { console: consolePlugin, application } = useKibana().services;
-
-  const { data: index } = useIndex(indexName);
+  const { console: consolePlugin, docLinks, application } = useKibana().services;
+  const { data: index, refetch, isLoading, isSuccess } = useIndex(indexName);
   const embeddableConsole = useMemo(
     () => (consolePlugin?.EmbeddableConsole ? <consolePlugin.EmbeddableConsole /> : null),
     [consolePlugin]
   );
   const navigateToIndexListPage = useCallback(() => {
     application.navigateToApp('management', { deepLinkId: 'index_management' });
   }, [application]);
+
+  const [showMoreOptions, setShowMoreOptions] = useState<boolean>(false);
+  const [isShowingDeleteModal, setShowDeleteIndexModal] = useState<boolean>(false);
+  const moreOptionsPopover = (
+    <EuiPopover
+      isOpen={showMoreOptions}
+      closePopover={() => setShowMoreOptions(!showMoreOptions)}
+      button={
+        <EuiButtonIcon
+          iconType="boxesVertical"
+          onClick={() => setShowMoreOptions(!showMoreOptions)}
+          size="m"
+          data-test-subj="moreOptionsActionButton"
+          aria-label={i18n.translate('xpack.searchIndices.moreOptions.ariaLabel', {
+            defaultMessage: 'More options',
+          })}
+        />
+      }
+    >
+      <EuiContextMenuPanel
+        data-test-subj="moreOptionsContextMenu"
+        items={[
+          <EuiContextMenuItem
+            key="trash"
+            icon={<EuiIcon type="trash" color="danger" />}
+            onClick={() => {
+              setShowDeleteIndexModal(!isShowingDeleteModal);
+            }}
+            size="s"
+            color="danger"
+            data-test-subj="moreOptionsDeleteIndex"
+          >
+            <EuiText size="s" color="danger">
+              {i18n.translate('xpack.searchIndices.moreOptions.deleteIndexLabel', {
+                defaultMessage: 'Delete Index',
+              })}
+            </EuiText>
+          </EuiContextMenuItem>,
+        ]}
+      />
+    </EuiPopover>
+  );
+
+  if (isLoading && !index) {
```

Comment:
Not for sure if it fits better but there is also `isInitialLoading` which should only be true for the 1st request and you wont have to do the `&& !index` here.

Reply:
updated to use `isInitialLoading` instead of `isLoading` in 5dde7ca7226ddcf1e89c639b17d6b16ebc34eba7